File: Arrays/BinarySearch.java
public class BinarySearch {
    public int binarySearch(int[] arr, int key, int low, int high) {

        if (low > high)
            return -1;

        int mid = (low + high) / 2;

        if (arr[mid] == key) return mid;
        else if (arr[mid] > key) return binarySearch(arr, key, low, mid - 1);
        else return binarySearch(arr, key, mid + 1, high);
    }
}

File: Arrays/BinarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

int* arr;

bool binarySearch(int aim, int l, int u)
{
    if(l>u) return false;
    int mid = (l+u)/2;

    if(arr[mid]==aim) return true;
    else if(arr[mid]>aim) return binarySearch(aim, l, mid-1);
    else return binarySearch(aim, mid+1, u);
}